### PR TITLE
[MBL-721] Remote Config Local Feature Flags

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Now that we have RemoteConfig configured, lets get our local feature flags in our debug menu working. 

# 🤔 Why

Some background context on why the change is needed.

# 🛠 How

These toggles will first reference values stored in user defaults, if there is one, before referencing the values currently published in our [firebase console](https://console.firebase.google.com/u/0/project/kickstarter-ios-alpha-94cf4/config).
Allowing us to test our app with different flags enbaled/disabled without impacting production.

We're using the same pattern here that we've had with Optimizely, so this is mostly renames and making sure that we're referencing the `RemoteConfigClientType` that is now in `AppEnvironment`

# 👀 See

Trello, screenshots, external resources?

| After 🦋 |
| --- |
|  Simulator Screen Recording - iPhone 13 Pro - 2023-05-08 at 14 05 28 |

# ✅ Acceptance criteria

Make sure you run the app using our `AppCenter Alpha` scheme to test.
- [x] Toggling flags in Debug -> Remote Config Feature Flags persists on relaunch and fully closing the app.
- [x] On a fresh install with Consent Management Dialog flag enabled, and the live value in the [firebase console](https://console.firebase.google.com/u/0/project/kickstarter-ios-alpha-94cf4/config) set to false, the app shows the Tracking dialog proving that the app is referencing the local flag value and not the published on

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by msadoon — CHANGES_REQUESTED** [review]
Ok, nice quick work here.

For some reason I'm not getting the flags to show in-app:

https://github.com/kickstarter/ios-oss/assets/4282741/cb561a2e-5a0f-49b0-8aae-ac9b70d486f8

Also address the questions below and I'll take another look, likely on Thursday. I want to focus heads down and get the xcode upgrade ready for review tomorrow.

File: Kickstarter-iOS/Features/BetaTools/Controller/BetaToolsViewController.swift [hunk 1]
```diff
@@ -166,7 +166,7 @@ internal final class BetaToolsViewController: UITableViewController {
   }
 
   private func goToOptimizelyFeatureFlagTools() {
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Kickstarter-iOS/Features/BetaTools/Controller/BetaToolsViewController.swift:168]: nice catch

**Comment by scottkicks** [comment]
addressed your comments @msadoon take your time reviewing. no rush!

**Review by msadoon — APPROVED** [review]
Yep so good to go  - I noticed that the defaults load first not the dashboard values (on fresh install). Then re-launching picks up the dashboard values. Just as a discussion this seems acceptable to me, but let me know if you think dashboard values should be shown on first install launch instead of default values.

**Pull request merged**